lib/UserAgentApplication.ts

```diff
@@ -193,12 +180,18 @@ namespace Msal {
                 this._tokenReceivedCallback = tokenReceivedCallback;
             }
 
-            this.redirectUri = window.location.href.split("?")[0].split("#")[0];
-            this.postLogoutredirectUri = this.redirectUri;
+            this._redirectUri = redirectUri || window.location.href.split("?")[0].split("#")[0];
+            this._postLogoutredirectUri = postLogoutRedirectUri || redirectUri;
+            this._navigateToLoginRequestUrl = navigateToLoginRequestUrl || true;
             this._loginInProgress = false;
             this._acquireTokenInProgress = false;
             this._renewStates = [];
             this._activeRenewals = {};
+            if (cacheLocation && !this._cacheLocations[cacheLocation]) {
```

Comment:
Should this logic be abstracted in a method to be called as either a setter or a default parameter value?